File: SubmitInstance/InstanceReader.cs
namespace SubmitInstance
{
	internal class InstanceReader : IInstanceReader
	{
		public string Read(string xmlFileName)
		{
			if (File.Exists(xmlFileName))
				using (StreamReader r = new(xmlFileName))
					return r.ReadToEnd();
			else
			{
				Console.WriteLine($"File {xmlFileName} cannot be found.");
				return string.Empty;
			}
		}
	}
}

File: VB/SubmitInstance/InstanceReader.vb
Imports System.IO

Friend Class InstanceReader
	Implements IInstanceReader

	Public Function Read(xmlFileName As String) As String Implements IInstanceReader.Read
		Dim absolutePath As String
		If Path.IsPathRooted(xmlFileName) Then
			absolutePath = xmlFileName
		Else
			absolutePath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, xmlFileName)
		End If

		If File.Exists(absolutePath) Then
			Dim r As StreamReader = New StreamReader(absolutePath)
			Return r.ReadToEnd()
		Else
			Return String.Empty
		End If
	End Function
End Class
